pandas/tools/merge.py

```diff
@@ -199,10 +204,53 @@ def get_result(self):
         typ = self.left._constructor
         result = typ(result_data).__finalize__(self, method='merge')
 
+        if self.indicator:
+            result = self._indicator_post_merge(result, indicator_string)
+
         self._maybe_add_join_keys(result, left_indexer, right_indexer)
 
         return result
 
+    def _indicator_pre_merge(self, left, right, indicator_arg):
+
+        if indicator_arg is True:
+            indicator_string = '_merge'
+        elif isinstance(indicator_arg, str):
+            indicator_string = indicator_arg
+        else:
+            raise ValueError('indicator option can only accept boolean or string arguments')
+        
```

Comment:
you don't need this validation at all, this will only be called ``if self.indicator``, so no problem